toolkit/smart-contracts/offchain/src/reserve/mod.rs

```diff
@@ -113,6 +126,33 @@ impl ReserveData {
 
 		Ok(ReserveUtxo { utxo: reserve_utxo, datum: reserve_settings })
 	}
+
+	pub(crate) async fn get_illiquid_circulation_supply_utxo<T: QueryLedgerState>(
+		&self,
+		ctx: &TransactionContext,
+		client: &T,
+	) -> Result<OgmiosUtxo, anyhow::Error> {
+		let validator_address = self
+			.scripts
+			.illiquid_circulation_supply_validator
+			.address(ctx.network)
+			.to_bech32(None)?;
+		let validator_utxos = client.query_utxos(&[validator_address]).await?;
+
+		let auth_token_asset_id = AssetId {
+			policy_id: self.scripts.illiquid_circulation_supply_auth_token_policy.policy_id(),
+			asset_name: AssetName::empty(),
+		};
+
+		let ics_utxo = validator_utxos
+			.into_iter()
+			.find(|utxo| utxo.get_asset_amount(&auth_token_asset_id) == 1u64)
+			.ok_or_else(|| {
+				anyhow!("Reserve Utxo not found, is the Reserve Token Management initialized?")
```

Comment:
```suggestion
				anyhow!("Could not find any UTXO with ICS Auth token at ICS Validator, is the Reserve Token Management initialized?")
```
Yes, I know I use "ICS Auth token" - I still think we should have some better name for it. But this renaming can wait. 

For sure "Reserve Utxo" is a name we use for the UTXO with Reserve Settings.